src/App.tsx

```diff
@@ -1,204 +1,187 @@
 /* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import classNames from 'classnames';
+
 import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import { getTodos, USER_ID } from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoCard } from './components/TodoCard';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [title, setTitle] = useState<string>('');
+  const [errorMessage, setErrorMessage] = useState<string>('');
+
+  const [allTodos, setAllTodos] = useState<boolean>(true);
+  const [activeTodos, setActiveTodos] = useState<boolean>(false);
+  const [completedeTodos, setCompletedTodos] = useState<boolean>(false);
+
+  useEffect(() => {
+    getTodos().then(setTodos);
+  }, []);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
+  const handleTitleSubmit = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setErrorMessage('');
+    setTitle(event.target.value);
+  };
+
+  const handleSubmit = () => {
+    if (!title) {
+      setErrorMessage('Title should not be empty');
+    }
+  };
+
+  const handleButtonClose = () => {
+    setErrorMessage('');
+  };
+
+  const filterTodoAll = () => {
```

Comment:
You shouldn't fetch todos from the server on every filter change. Instead fetch it once (as you did in useEffect). on filter click save the filter value to a separate state.

Then having all `todos` and `filter` status value, you can filter todos depending on status and save filtered todos to a common variable. On filter or todos state change rerender will happen and this common variable will be recomputed.

Reply:
P.S. use enum for Filter values